src/Rules/LazyRegistry.php

```diff
@@ -32,7 +32,17 @@ public function __construct(private Container $container)
 	public function getRules(string $nodeType): array
 	{
 		if (!isset($this->cache[$nodeType])) {
-			$parentNodeTypes = [$nodeType] + class_parents($nodeType) + class_implements($nodeType);
+			$parents = class_parents($nodeType);
```

Comment:
Why is this piece of code changed? ClassImplementsFunctionReturnTypeExtension makes sure the returned value does not include `false`.

Reply:
It got reported by `make phpstan`. I'll take another look.

Reply:
There are two errors reported:

```
Binary operation "+" between non-empty-array and (array<string, class-string>|false) results in an error.
Binary operation "+" between array{class-string<TNodeType of PhpParser\Node>} and (array<string, class-string>|false) results in an error.
```

The problem is this (in `InvalidBinaryOperationRule`):
`$callback = static fn (Type $type): bool => !$type->toNumber() instanceof ErrorType;`

`ClassImplementsFunctionReturnTypeExtension` returns benevolent union of array and false. And the callback only keeps false. So the rule checks `array + false`, which is an error.

I'll try to fix it.